src/options/App.tsx

```diff
@@ -93,57 +92,64 @@ function OptionsPage(props: { theme: Theme; onThemeChange: (theme: Theme) => voi
       <main className="w-[600px] mx-auto mt-14">
         <Text h2>Options</Text>
         <Text h3 className="mt-5">
-          Prompt
+          Site Config
         </Text>
-
-        <PromptCard
-          header={'default'}
-          onSave={(prompt) => updateUserConfig({ prompt })}
-          prompt={prompt}
-        />
-
-        {promptOverrides.map((override, index) => {
-          return (
-            <div key={override.site} className="my-3">
-              <PromptCard
-                header={override.site}
-                prompt={override.prompt}
-                onSave={(newPrompt) => {
-                  const newOverride: SitePrompt = {
-                    site: override.site,
-                    prompt: newPrompt,
-                  }
-                  const newOverrides = promptOverrides.filter((o) => o.site !== override.site)
-                  newOverrides.splice(index, 0, newOverride)
-                  setPromptOverrides(newOverrides)
-                  return updateUserConfig({ promptOverrides: newOverrides })
-                }}
-                onDismiss={() => {
-                  const newOverrides = promptOverrides.filter((_, i) => i !== index)
-                  setPromptOverrides(newOverrides)
-                  return updateUserConfig({ promptOverrides: newOverrides })
-                }}
-              />
-            </div>
-          )
-        })}
+        {supportedURLs &&
+          supportedURLs.map((URL, index) => {
+            return (
+              <div key={URL} className="my-3">
+                <SiteCard
+                  prompt={siteConfigs[URL].prompt}
+                  URL={URL}
+                  bodyTag={siteConfigs[URL].bodyTag}
+                  displayTag={siteConfigs[URL].displayTag}
+                  onSave={(URL, prompt, bodyTag, displayTag) => {
+                    supportedURLs[index] = URL
+                    siteConfigs[URL] = {
+                      prompt: prompt,
+                      bodyTag: bodyTag,
+                      displayTag: displayTag,
+                      URL: URL,
+                    }
+                    return updateUserConfig({
+                      supportedURLs: supportedURLs,
+                      siteConfigs: siteConfigs,
+                    })
+                  }}
+                  onDismiss={() => {
+                    supportedURLs.splice(index, 1)
```

Comment:
Please keep objects immutable and create a new object instead of updating old one.